libraries/standard/mqtt/src/iot_mqtt_network.c

```diff
@@ -483,13 +483,18 @@ static IotMqttError_t _deserializeIncomingPacket( _mqttConnection_t * pMqttConne
                                                   _mqttPacket_t * pIncomingPacket )
 {
     IotMqttError_t status = IOT_MQTT_STATUS_PENDING;
+    bool packetValidity = false;
+
+    /* Packet validity is not checked when asserts are disabled. */
+    ( void ) packetValidity;
 
     /* A buffer for remaining data must be allocated if remaining length is not 0. */
     IotMqtt_Assert( ( pIncomingPacket->remainingLength > 0 ) ==
                     ( pIncomingPacket->pRemainingData != NULL ) );
 
     /* Only valid packets should be given to this function. */
-    IotMqtt_Assert( _incomingPacketValid( pIncomingPacket->type ) == true );
```

Comment:
The 13.6 violation here was "The operand of the sizeof operator contains potential side effect: calling function _incomingPacketValid(pIncomingPacket->type)". This seems to be a false positive as the `sizeof` operator shouldn't be getting called. I introduced a boolean here to reduce the number of coverity exceptions we have in comments, but that can be changed if it's more desirable.

Reply:
Should change it back.  `packetValidity` is not used when asserts are disabled.

Reply:
Reverted